src/gebt_poc/mesh.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+namespace openturbine::gebt_poc {
+class Mesh {
+public:
+    int GetNumberOfElements() const { return number_of_elements_; }
+
+    int GetNumberOfNodes() const { return number_of_nodes_; }
+
+    KOKKOS_FUNCTION
+    Kokkos::View<const int*> GetNodesForElement(int elementID) const {
+        return Kokkos::subview(connectivity_, elementID, Kokkos::ALL);
+    }
+
+    friend Mesh create1DMesh(int number_of_elements, int nodes_per_element);
+
+protected:
+    Mesh() = default;
+    void SetNumberOfElements(int number_of_elements) { number_of_elements_ = number_of_elements; }
+    void SetNumberOfNodes(int number_of_nodes) { number_of_nodes_ = number_of_nodes; }
+
+    void InitializeElementNodeConnectivity(int number_of_elements, int nodes_per_element) {
+        connectivity_ = Kokkos::View<int**>("connectivity", number_of_elements, nodes_per_element);
+    }
+
+    void CheckElementConsistency() {
+        if (static_cast<unsigned>(number_of_elements_) != connectivity_.extent(0)) {
```

Comment:
Let's do the type casting to `std::size_t` to be consistent with rest of the code in this file.
`if (static_cast<std::size_t>(number_of_elements_) != connectivity_.extent(0)) {`